appveyor.yml

```diff
@@ -105,6 +105,8 @@ build_script:
   - python -m pip install --user --upgrade pip setuptools setuptools-scm wheel || exit
   - python -m pip install --no-cache-dir --user -r bindings/py/packaging/requirements.txt  || exit
 
+  # we need to find the previous git tags so a shallow clone is not enough
+  - git fetch --unshallow
```

Comment:
I think this clones the whole repo, which is unnecessary and takes time. 
https://stackoverflow.com/questions/6802145/how-to-convert-a-git-shallow-clone-to-a-full-clone
`git fetch --depth=N` for some reasonable size, say 200 commits between each release?

Reply:
Oh, good idea.   Do you think 200 commits is enough?
I will try 200 and if we find we are going longer between tags we can bump that.